CedarJava/src/test/java/com/cedarpolicy/pbt/ActionGen.java

```diff
@@ -31,21 +32,24 @@ public final class ActionGen {
     /** Generate a random Action Group. */
     public static List<Entity> getEntities() {
         List<Entity> actions = new ArrayList<>();
-        String actionEuid = "Action::\"" + Utils.strings() + "\"";
+        String actionType = "Action";
+        String actionId = Utils.strings();
+        String actionEuid = actionType+"::\"" + actionId + "\"";
         Map<String, Value> actionAttributes = new HashMap<>();
-        Set<String> actionParents = new HashSet<>();
-        Entity e = new Entity(actionEuid, actionAttributes, actionParents);
+        Set<JsonEUID> actionParents = new HashSet<>();
+        Entity e = new Entity(new JsonEUID(actionType, actionId), actionAttributes, actionParents);
         actions.add(e);
         int count = Arbitraries.integers().between(10, 100).sample();
 
         for (int i = 0; i < count; i++) {
-            actionEuid = "Action::\"" + Utils.strings() + "\"";
-            if (!e.uid.equals(actionEuid)) {
-                e.parents.add(actionEuid);
+            actionId = Utils.strings();
+            actionEuid = "Action::\"" + actionId + "\"";
+            if (!e.getEuid().toCedar().equals(actionEuid)) {
+                e.parentsEUIDs.add(new JsonEUID(actionType, actionId));
```

Comment:
We can't use the default `equals` method here and if I define `equals` SpotBugs says I should also define `hashCode`. Since I don't want to do that and string equality is exactly what we want, I think I'll leave as-is. I did rename `toCedar` -> `toString` and removed the old `toString` though.

Reply:
Sounds good. What about the EUID vs. Euid issue?

Reply:
Let's change that in a separate PR since it will touch a bunch of files.

Reply:
I found https://stackoverflow.com/questions/15526107/acronyms-in-camelcase which gives various options. I personally found this answer (https://stackoverflow.com/a/45440841) most reasonable. Unfortunately, it violates Jackson's naming convention (JavaScript Object Notation could be argued to go to `JSON` or `JsON` but not `Json`). It would also indicate Entity Unique Identifier should go to `EUId` which 1) looks weird to me and 2) doesn't match our docs.

I put up (https://github.com/cedar-policy/cedar-java/pull/42) with some obvious consistency changes, but I'm not convinced which style is best for us, so it's only `Euid -> EUID`